crates/cheatnet/src/state.rs

```diff
@@ -233,7 +247,31 @@ impl CheatnetState {
 
     #[must_use]
     pub fn address_is_warped(&self, contract_address: &ContractAddress) -> bool {
-        self.warped_contracts.contains_key(contract_address)
+        self.global_warp.is_some()
+            || matches!(
+                self.warped_contracts.get(contract_address),
+                Some(WarpedContract::Warped(_))
+            )
+    }
+
+    #[must_use]
+    pub fn get_address_block_timestamp(&self, address: &ContractAddress) -> Option<Felt252> {
+        /* The Logic
+         * - If there is a global warp and no contract warp, use the global warp
+         * - If there is a global warp and a contract warp (either Warped or Unwarped), use the contract warp
+         * - If there is no global warp and a contract warp, use the contract warp (either Warped or Unwarped)
+         * - If there is no global warp and no contract warp, return None (no warp)
+         *
+         * This logic ensures that `start_warp` and `stop_warp` will always behave as expected,
+         */
+
+        match (&self.global_warp, self.warped_contracts.get(address)) {
+            (_, Some(warped_contract)) => match warped_contract {
+                WarpedContract::Warped(contract_timestamp) => Some(contract_timestamp.clone()),
+                WarpedContract::Unwarped => None,
+            },
+            (any_global, None) => any_global.clone(),
+        }
```

Comment:
I think we can express the comment above with a bit more expressive code thus removing the need to comment entirely.
```suggestion
if let Some(warped_contract) = self.warped_contracts.get(address) {
    match warped_contract {
           WarpedContract::Warped(contract_timestamp) => Some(contract_timestamp.clone()),
            WarpedContract::Unwarped => None,
        },
} else {
    self.global_warp
}
```